### PR TITLE
Add information on contributing documentation

### PR DESCRIPTION
**Pull request by yukiisbored**

### CODE REVIEW

**Comment by yukiisbored** [comment]
Not entirely sure if this section is good enough as it doesn't really point the reader to information on how to actually contribute documentation.

Detailed project-specific information is currently not available:
- https://github.com/NixOS/nix/issues/7204
- https://github.com/NixOS/nixpkgs/issues/197233

**Comment by yukiisbored** [comment]
Seems like linkcheck fails on GitHub markdown anchors.

File: source/contributing.md [hunk 1]
```diff
@@ -62,10 +79,15 @@ Feel free to [join our community] of developers!
...
 [nixpkgs]: https://github.com/NixOS/nixpkgs
 [programming language specific instructions]: https://nixos.org/manual/nixpkgs/stable/#chap-language-support
-[the manual]: https://nixos.org/manual/nix/stable/quick-start.html
+[nix manual]: https://nixos.org/manual/nix/stable/quick-start.html
+[nixpkgs manual]: https://nixos.org/manual/nixpkgs/stable
+[nixos manual]: https://nixos.org/manual/nixos/stable
 [nix issues]: https://github.com/NixOS/nix/issues
 [nixos issues]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%226.topic%3A+nixos%22
 [nixpkgs issues]: https://github.com/NixOS/nixpkgs/issues?page=3&q=is%3Aissue+is%3Aopen+label%3A%228.has%3A+package+%28update%29%22%2C%228.has%3A+package+%28new%29%22%2C%229.needs%3A+package+%28update%29%22%2C%229.needs%3A+package+%28new%29%22%2C%220.kind%3A+packaging+request%22%2C%220.kind%3A+build+failure%22
+[nix.dev]: https://nix.dev
```
Comment (fricklerhandwerk) [inline source/contributing.md:91]: Would be better to have the links closer to usage, it's easier to find.

File: source/contributing.md [hunk 2]
```diff
@@ -20,10 +20,27 @@ as they help us solve the problem.
...
+
+[nix.dev] is a curated and maintained repository of guides and tutorials aimed
+to teach newcomers essential Nix knowledge, show best practices, and help orient
+users in the Nix ecosystem.
+
+[General guidelines on writing documentation] is available in the nix.dev
+repository.
+
+[Detailed guidelines on contributing documentation] is available in the Nix
+documentation team repository.
```
Comment (fricklerhandwerk) [inline source/contributing.md:38]:
```suggestion
[General guidelines on writing documentation] are available in the nix.dev
repository.

[Detailed guidelines on contributing documentation] are available in the Nix
documentation team repository.
```

**Review by fricklerhandwerk — APPROVED** [review]
Just a few small things. Looks good, thanks a lot!

File: source/contributing.md [hunk 3]
```diff
@@ -20,6 +20,32 @@ as they help us solve the problem.
 
 You need a [GitHub] account for that.
 
+[github]: https://github.com/
+
+## Write documentation
+
+Documentations on the Nix ecosystem are separated by their purpose.
```
Comment (lucperkins) [inline source/contributing.md:27]: Documentation should never be pluralized. In this case, I'd recommend something like: "Documentation on the Nix ecosystem is divided into several domains based on purpose."

File: source/contributing.md [hunk 4]
```diff
@@ -32,24 +58,31 @@ issues][nix issues].
 Feel free to [join our community] to get in
 contact with other developers.
 
+[how to setup a development environment]: https://nixos.org/manual/nix/stable/contributing/hacking.html
```
Comment (Mic92) [inline source/contributing.md:61]: Maybe link to unstable manual to point to the latest docs.

File: source/contributing.md [hunk 5]
```diff
@@ -20,6 +20,32 @@ as they help us solve the problem.
...
 You need a [GitHub] account for that.
 
+[github]: https://github.com/
+
+## Write documentation
+
+Documentations on the Nix ecosystem are separated by their purpose.
+
+The manuals for [Nix][nix manual], [Nixpkgs][nixpkgs manual], and [NixOS][nixos
+manual] are reference documentation, specifying interface and behavior.
```
Comment (fricklerhandwerk) [inline source/contributing.md:30]:
```suggestion
The manuals for [Nix][nix manual], [Nixpkgs][nixpkgs manual], and [NixOS][nixos manual] are reference documentation, specifying interfaces and behavior. 
```

File: source/contributing.md [hunk 6]
```diff
@@ -20,6 +20,32 @@ as they help us solve the problem.
...
+
+## Write documentation
+
+Documentations on the Nix ecosystem are separated by their purpose.
+
+The manuals for [Nix][nix manual], [Nixpkgs][nixpkgs manual], and [NixOS][nixos
+manual] are reference documentation, specifying interface and behavior. 
+
+[nix.dev] is a curated and maintained repository of guides and tutorials aimed
+to teach newcomers essential Nix knowledge, show best practices, and help orient
```
Comment (fricklerhandwerk) [inline source/contributing.md:33]: maybe
```suggestion
to teach beginners essential Nix knowledge, show best practices, and help orient
```

File: source/contributing.md (hunk 5 above)
Comment (fricklerhandwerk) [reply source/contributing.md:30]: Also please link to sources.

**Review by fricklerhandwerk — APPROVED** [review]
Please use one line per sentence.

File: source/contributing.md [hunk 7]
```diff
@@ -20,52 +20,81 @@ as they help us solve the problem.
...
+[github]: https://github.com/
+
+## Contribute to documentation
+
+Documentation on the Nix ecosystem is divided into several domains based on purpose.
+
+The manuals for [Nix][nix manual] ([src][nix manual src]), [Nixpkgs][nixpkgs
+manual] ([src][nixpkgs manual src]), and [NixOS][nixos manual] ([src][nixos manual src]) 
+are reference documentation, specifying interfaces and behavior.
+
```
Comment (fricklerhandwerk) [inline source/contributing.md:32]:
```suggestion
```

This information is already in the general guidelines.

File: source/contributing.md [hunk 8]
```diff
@@ -20,52 +20,77 @@ as they help us solve the problem.
...
+[join our community]: https://github.com/NixOS/nixpkgs#community
+
+## Contribute to Nixpkgs
 
 Packaging for Nix is simple when you have understood the basic concept.
 
-[The manual] explains step-by-step how to add new packages to the Nix
-package collection. There are also [programming language specific
-instructions][programming language specific instructions].
+[The manual][nixpkgs manual] explains step-by-step how to add new packages to
```
Comment (fricklerhandwerk) [inline source/contributing.md:69]: This has to link to the contributing section

**Review by fricklerhandwerk — APPROVED** [review]

**Review by fricklerhandwerk — APPROVED** [review]

**Comment by fricklerhandwerk** [comment]
Great job, thanks a lot for all the fixups!

**Pull request merged**